src/OpenTelemetry/BatchExportProcessor.cs

```diff
@@ -110,6 +111,14 @@ public class BatchExportProcessor<T> : BaseExportProcessor<T>
         /// <inheritdoc/>
         public override void OnExport(T data)
         {
+            lock (this.exporterProcException)
+            {
+                if (this.exporterProcException != null)
```

Comment:
Is this nested check necessary?

Reply:
You're right. It's not necessary. I'll be getting rid of the lock statement as we could log the exception on the thread that runs ExporterProc().